moveit_ros/moveit_servo/src/collision_check.cpp

```diff
@@ -80,6 +80,12 @@ CollisionCheck::CollisionCheck(ros::NodeHandle& nh, const moveit_servo::ServoPar
 
   current_state_ = planning_scene_monitor_->getStateMonitor()->getCurrentState();
   acm_ = getLockedPlanningSceneRO()->getAllowedCollisionMatrix();
+
+  // Set up Bullet collision detection
+  collision_det_allocation_ = std::make_shared<collision_detection::CollisionDetectorAllocatorBullet>();
+  const collision_detection::WorldPtr& world =
+      planning_scene_monitor::LockedPlanningSceneRW(planning_scene_monitor_)->getWorldNonConst();
+  collision_env_ = collision_det_allocation_->allocateEnv(world, getLockedPlanningSceneRO()->getRobotModel());
```

Comment:
Won't this just take an initial snapshot of the collision world so that changes in the scene aren't handled at all? I definitely agree that we should decouple collision checks from the PSM. One way to achieve this would be to create a decoupled child scene that we keep updated for any changes in the parent. What do you think?

Reply:
Transforms are updated but (I believe) new objects aren't detected. I'll test it.

Obviously I'd rather avoid copying the planning scene if this simple approach works.

Reply:
If this approach works (and doesn't allow race conditions) I'm perfectly fine with this.